components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/internal/SAMLApplicationMgtListener.java

```diff
@@ -186,16 +185,25 @@ public void doImportServiceProvider(ServiceProvider serviceProvider) throws Iden
                         .getInboundAuthenticationRequestConfigs()) {
                     if (StringUtils.equals(authConfig.getInboundAuthType(), SAMLSSO)) {
 
+                        SAMLSSOServiceProviderDTO samlssoServiceProviderDTO =
+                                (SAMLSSOServiceProviderDTO) authConfig.getInboundConfigurationProtocol();
+
```

Comment:
shall we try and refactor the method `onPreCreateInbound`? The method seems to be too long and the control flow is not very clear

Reply:
Refactored by adding a separate ```getSAMLSSOServiceProviderDTO``` method.